### PR TITLE
Add instance and server reconfigure support

### PR DESCRIPTION
**Pull request by nick-phillips-hpe**

### CODE REVIEW

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy [hunk 1]
```diff
@@ -1020,6 +1030,227 @@ class NutanixPrismElementPluginProvisionProvider extends AbstractProvisionProvid
...
+		if (resizeResults.success) {
+			def computeServer = context.services.computeServer.get(server.id)
+			computeServer.coresPerSocket = resizeRequest.coresPerSocket
+			computeServer.maxCores = resizeRequest.maxCores
+			computeServer.maxMemory = resizeRequest.maxMemory
+			context.services.computeServer.save(computeServer)
+			return rtn
+		}
+
+		return rtn
```
Comment (mcarlinhpe) [inline src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: This looks like it always returns success even if `resizeResults` contains a failure. Should this propagate a failure up? If not, can we add a comment?

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy [hunk 2]
```diff
@@ -1020,6 +1030,227 @@ class NutanixPrismElementPluginProvisionProvider extends AbstractProvisionProvid
...
+				def dataStore = getDatastoreOption(cloud, server.account, null, null, it.size.toLong())
+				addOpts.containerId = dataStore?.externalId
+			}
+			def sizeGb = it.size?.toInteger()
+			def busType = context.services.storageVolume.storageVolumeType.get(it?.storageType?.toLong())?.code?.replace('nutanix-','')
+			def diskResults = NutanixPrismElementApiService.addDisk(client, addOpts, vmId, sizeGb, busType)
+			log.debug("create disk success: ${diskResults.success}")
+			if(!diskResults.success) {
+				log.error "Error in creating the: ${diskResults}"
+				rtn.error = "Error in creating the"
```
Comment (mcarlinhpe) [inline src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: unfinished error message?

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy [hunk 3]
```diff
@@ -1020,6 +1030,227 @@ class NutanixPrismElementPluginProvisionProvider extends AbstractProvisionProvid
...
+			if(deleteResults.success == true) {
+				def networkInterface = context.services.computeServer.computeServerInterface.get(networkDelete.id)
+				computeServer.interfaces.remove(networkInterface)
+				// TODO: switch back to bulkRemove once fixed
+				context.async.computeServer.computeServerInterface.remove([networkDelete], computeServer).blockingGet()
+			}
+		}
+
+		context.services.computeServer.save(computeServer)
+		return rtn
```
Comment (mcarlinhpe) [inline src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: Same thing here. Should this propagate a failure up? If not, can we add a comment?

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy (hunk 1 above)
Comment (nick-phillips-hpe) [reply src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: service response should have been init with "prepare". will fix

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy (hunk 2 above)
Comment (nick-phillips-hpe) [reply src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: idk this was copy pasta from the embedded. will finish the sentence

File: src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy (hunk 3 above)
Comment (nick-phillips-hpe) [reply src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy]: will add comment

**Review by ddeyoung — APPROVED** [review]

**Pull request merged**